File: java/src/main/java/com/workshop/architecture/fitness/inside/port/outbound/MembershipBillingReference.java
package com.workshop.architecture.fitness.inside.port.outbound;

import java.time.Instant;
import java.time.LocalDate;
import java.util.UUID;

public record MembershipBillingReference(UUID id,
                                         UUID membershipId,
                                         String externalInvoiceId,
                                         String externalInvoiceReference,
                                         LocalDate dueDate,
                                         String status,
                                         Instant createdAt,
                                         Instant updatedAt) {
}

File: python/src/workshop_api/fitness/hexagon/inside/port/outbound/membership_billing_reference.py
from dataclasses import dataclass
from datetime import date, datetime
from uuid import UUID


@dataclass(frozen=True)
class MembershipBillingReference:
    id: UUID
    membership_id: UUID
    external_invoice_id: str
    external_invoice_reference: str
    due_date: date
    status: str
    created_at: datetime
    updated_at: datetime
